pubnub-gson/src/main/java/com/pubnub/internal/v2/subscription/SubscriptionImpl.kt

```diff
@@ -36,6 +50,10 @@ class SubscriptionImpl(
         return pubnub.subscriptionSetOf(setOf(this, subscription))
     }
 
+    override fun subscribe() {
+        super.subscribe(SubscriptionCursor(0))
```

Comment:
why super? just `subscribe(SubscriptionCursor(0))`
if we ever override it in this class, the no-argument function should use the override, not the superclass implementation


this applies to other places in this PR where `super.XXX` is called

Reply:
For me:
super.subscribe(SubscriptionCursor(0))
is safer and straightforward. It protects from accidental recursion. We can always change it to 
subscribe(SubscriptionCursor(0)) in the future when there is a need to override subscribe(cursor: SubscriptionCursor) in this class.

What do you think?

Reply:
the problem is that the compiler won't tell someone in the future that this needs to be changed...
and I don't think we're gaining anything now by calling `super` other than potential problems later.
can you explain why it's safer?

even if you look at the code generated by kotlin @JvmOverloads (if we could use it here), it would always generate `this.subscribe(...)`, because that's just the desired behaviour that we're trying to achieve here.

Reply:
I think I see your point.

Problematic situation might be:
```
    override fun subscribe() {
        subscribe(SubscriptionCursor(0))
    }

    override fun subscribe(cursor: SubscriptionCursor) {
        subscribe(cursor)
    }

```
For me both approaches(using super or not) have pros and cons. 
I can go with your proposal.

Reply:
Other thing that I don't like is when there is:
`subscribe(SubscriptionCursor(0))`
and you click on subscribe method it shows two options.